_posts/2025-03-20-phased-strimzi-upgrade-example.md

```diff
@@ -0,0 +1,637 @@
+    target:
+      kind: (RoleBinding|ClusterRoleBinding)
+      name: ".*"
+```
+
+This patch will, for _all_ `RoleBinding` and `ClusterRoleBinding` in the `resources` list, replace the name of the first entry in the `subjects` list (in the base installation files this always the operator `ServiceAccount`) with the name specified in the patch. 
+This name should be the original name in the base Strimzi  operator `ServiceAccount` plus the `nameSuffix`.
+It will also patch the `namespace` which should match the namespace where the service account is installed. 
+In this case we assume all operator versions are installed in the `strimzi` namespace.
+
+#### Watching all namespaces
+
+The final thing we need to setup is configuring the namespaces that the Strimzi Operators will watch.
+For this example we assume that the various Strimzi versions will be watching all namespaces on the Kubernetes cluster.
+As per [the documentation](https://strimzi.io/docs/operators/latest/deploying#deploying-cluster-operator-to-watch-whole-cluster-str) we need to alter the `RoleBinding` and `ClusterRoleBinding` namespaces to match the one the operator is installed in.
+However, that is already handled by the step above.
+
+We then need to set the `STRIMZI_NAMESPACE` env var in the operator `Deployment` to `*`.
+For this we can update the `patch-deployment.yaml` from the earlier step:
+
+`install-files/cluster-operator-0.43.0/patches/patch-deployment.yaml`
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: strimzi-cluster-operator
+spec:
+  template:
+    spec:
+      containers:
+        - name: strimzi-cluster-operator
+          env:
+            - name: STRIMZI_NAMESPACE
+              value: "*" 
+              valueFrom: null
+            - name: STRIMZI_CUSTOM_RESOURCE_SELECTOR
+              value: strimzi-resource-selector=strimzi-0-43-0
+            - name: STRIMZI_LEADER_ELECTION_LEASE_NAME
+              value: "strimzi-cluster-operator-0-43-0"
+```
+
+[The documentation](https://strimzi.io/docs/operators/latest/deploying#deploying-cluster-operator-to-watch-whole-cluster-str) then says we need to create several additional (Cluster)Roles and bindings. 
+We can add these to the base install resources by using the `--dry-run=client` and `-o yaml` flags for the `kubectle create` command to output the resource files to yaml format and then send that to a file.
+For example:
+
+```shell
+kubectl create --dry-run=client -o yaml \
+  clusterrolebinding strimzi-cluster-operator-namespaced \
+  --clusterrole=strimzi-cluster-operator-namespaced \
+  --serviceaccount strimzi:strimzi-cluster-operator \
+  > install-files/cluster-operator-0.43.0/strimzi-cluster-operator-namespaced.yaml
+```
+
+We can then add these additional resources to the `resources` list in the respective `kustomization.yaml` files and they will be modified by the same `nameSuffix` and `patch-role-reference.yaml` patches as the other RBAC resources.
```

Comment:
Doesn't this just create a new installation file while keeping the old RoleBinding in place as well?

Reply:
Yes, but with this setup all the RBAC is completely seperate, different names pointing at different roles and different service accounts.